mlrun/projects/project.py

```diff
@@ -3821,6 +3821,27 @@ def list_functions(self, name=None, tag=None, labels=None):
             # convert dict to function objects
             return [mlrun.new_function(runtime=func) for func in functions]
 
+    def list_functions_by_foo_spec(self, foo_spec, tag=None, labels=None):
+        """Retrieve a list of functions, filtered by specific criteria.
+
+        example::
+
+            functions = project.list_functions_by_foo_spec(foo_spec="bar")
+
+
+        :param foo_spec: Return only functions with a specific foo_spec.
+        :param tag: Return function versions with specific tags. To return only tagged functions, set tag to ``"*"``.
+        :param labels: Return functions that have specific labels assigned to them.
+        :returns: List of function objects.
```

Comment:
General suggestion on docstring formatting
```suggestion
        Example::

            functions = project.list_functions_by_foo_spec(foo_spec="bar")


        :param foo_spec: Return only functions with a specific foo_spec.
        :param tag:      Return function versions with specific tags. To return only tagged functions, set tag to ``"*"``.
        :param labels:   Return functions that have specific labels assigned to them.

        :returns: List of function objects.
```